segmentstore/server/src/main/java/io/pravega/segmentstore/server/WriterSegmentProcessor.java

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.segmentstore.server;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Defines a general Segment Processor for the Storage Writer.
+ */
+public interface WriterSegmentProcessor extends AutoCloseable {
+
+    @Override
+    void close();
+
+    /**
+     * Gets a value indicating whether the SegmentAggregator is closed (for any kind of operations).
```

Comment:
This comment is referring to `SegmentAggregator`, but this is not about the segment aggregator alone.